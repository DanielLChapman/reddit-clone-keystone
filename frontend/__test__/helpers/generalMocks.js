import { CURRENT_USER_QUERY } from '../../components/User';
import { fakeTextPost, fakeUser} from '../../lib/testUtils';

const fakeUserObj = fakeUser();
const notSignedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: null } },
    },
  ];
  
const signedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: fakeUser() /*{
        
            id: fakeUserObj.id,
            username: fakeUserObj.username,
            email: fakeUserObj.email,
            name:fakeUserObj.name,
            posts: [],
            comments: [],
            moderating: [],
            owner:[],
            subreddits:[],
            postvotes:[],
            commentvotes: [],
        
      }*/ } },
    },
  ];
  
const signedInMocksWithPosts = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: {
        data: {
          authenticatedItem: fakeUser({
            posts: [fakeTextPost()],
          }),
        },
      },
    },
];

export {
    notSignedInMocks,
    signedInMocks,
    signedInMocksWithPosts
}

/* 

{
        id
        username
        email
        name
        posts {
          votes {
            vflag
          }
        }
        comments {
          votes {
            vflag
          }
        }
        moderating {
          id
          slug
          name
        }
        owner {
          id
          slug
          name
        }
        subreddits {
          id
          slug
          name
        }
        postvotes {
          post {
            title
            id
          }
          id
          vflag
        }
        commentvotes {
          comment {
            id
          }
          id
          vflag
        }
    }

    */