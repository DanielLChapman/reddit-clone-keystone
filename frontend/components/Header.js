import Link from 'next/link';
import Nav from './Nav';


export default function Header() {
  return (
    <>
      <div className="bar">
        <span>
          <Link href="/">App name</Link>
        </span>
        <Nav />
      </div>
    </>
  );
}
