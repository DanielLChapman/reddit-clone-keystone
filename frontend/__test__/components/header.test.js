import { act, render, screen } from '@testing-library/react';
import {MockedProvider} from '@apollo/react-testing';
import Header from '../../components/Header';
import { signedInMocks, signedInMocksWithPosts } from '../helpers/generalMocks';
import wait from 'waait';


describe('It insures the header contains the necessary information', () => {
    it('has a logo, search bar, and dropdown menu', () => {
        const { container, debug } = render(
            <MockedProvider >
              <Header />
            </MockedProvider>
          );

          expect(container).toMatchSnapshot();
    });

    it ('renders a different drop down when logged in ', async () => {
        const { container: container2, debug: debug2 } = render(
            <MockedProvider >
              <Header />
            </MockedProvider>
          );
        const { container, debug} = render(
            <MockedProvider mocks={signedInMocks}>
                <Header />
            </MockedProvider>
        );

        
        await act(async() => {
            await new Promise(resolve => setTimeout(resolve, 0));
        })

        expect(container).toMatchSnapshot();
        expect(container).not.toBe(container2);

    })
})