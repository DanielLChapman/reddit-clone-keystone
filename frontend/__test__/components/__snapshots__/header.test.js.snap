// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It insures the header contains the necessary information has a logo, search bar, and dropdown menu 1`] = `
<div>
  <div
    class="top-bear-container"
  >
    <div
      class="top-bar"
    >
      <a
        class="website-icon"
        href="/"
      >
        <img
          src="/icon.webp"
        />
        <span>
          Reddit
        </span>
      </a>
      <div
        class="search-bar-main"
      >
        <div
          class="search"
        >
          <button
            class="searchButton"
            type="submit"
          >
            <i
              class="fa fa-search"
            />
          </button>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="downshift-0-menu"
            role="combobox"
          >
            <input
              aria-autocomplete="list"
              aria-controls="downshift-0-menu"
              aria-labelledby="downshift-0-label"
              autocomplete="off"
              class="searchTerm"
              id="search"
              placeholder="Search"
              type="search"
              value=""
            />
          </div>
          <div
            aria-labelledby="downshift-0-label"
            class="Dropdown__DropDown-sc-1liqluw-0 TgajO dropdown-window dropdown-not-opened"
            data-testid="search-bar-results"
            id="downshift-0-menu"
            role="listbox"
          />
        </div>
      </div>
      <section
        class="signin-login-dropdown"
      >
        <label
          for="signin-login-window"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-w-16 fa-user-alt "
            data-icon="user-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"
              fill="currentColor"
            />
          </svg>
          <span
            style="top: 0px; position: relative;"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-w-14 fa-arrow-down "
              data-icon="arrow-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
                fill="currentColor"
              />
            </svg>
          </span>
        </label>
        <input
          id="signin-login-window"
          name="signin-login-window"
          type="checkbox"
        />
        <div>
          <ul />
          <button
            class="signup-button"
            style="top: -1px;"
            type="submit"
          >
            <a
              href="/signin"
            >
              Sign up
            </a>
          </button>
          <button
            class="dropdown-button"
          >
            <a
              href="/signin"
            >
              Sign In
            </a>
          </button>
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`It insures the header contains the necessary information renders a different drop down when logged in  1`] = `
<div>
  <div
    class="top-bear-container"
  >
    <div
      class="top-bar"
    >
      <a
        class="website-icon"
        href="/"
      >
        <img
          src="/icon.webp"
        />
        <span>
          Reddit
        </span>
      </a>
      <div
        class="search-bar-main"
      >
        <div
          class="search"
        >
          <button
            class="searchButton"
            type="submit"
          >
            <i
              class="fa fa-search"
            />
          </button>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="downshift-0-menu"
            role="combobox"
          >
            <input
              aria-autocomplete="list"
              aria-controls="downshift-0-menu"
              aria-labelledby="downshift-0-label"
              autocomplete="off"
              class="searchTerm"
              id="search"
              placeholder="Search"
              type="search"
              value=""
            />
          </div>
          <div
            aria-labelledby="downshift-0-label"
            class="Dropdown__DropDown-sc-1liqluw-0 TgajO dropdown-window dropdown-not-opened"
            data-testid="search-bar-results"
            id="downshift-0-menu"
            role="listbox"
          />
        </div>
      </div>
      <section
        class="signin-login-dropdown"
      >
        <label
          for="signin-login-window"
        >
          <span
            class="signin-login-window-username"
          >
            Ms. Josue McCullough
          </span>
          <span
            style="top: 0px; position: relative;"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-w-14 fa-arrow-down "
              data-icon="arrow-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
                fill="currentColor"
              />
            </svg>
          </span>
        </label>
        <input
          id="signin-login-window"
          name="signin-login-window"
          type="checkbox"
        />
        <div>
          <ul>
            <li>
              <a
                href="/user/Ms. Josue McCullough"
              >
                Profile
              </a>
            </li>
            <li>
              <span
                style="float: left;"
              >
                Karma
              </span>
                
              <span
                style="float: right;"
              >
                0
              </span>
              <br />
            </li>
            <li>
              <a
                href="/subreddits/create"
              >
                Create Subreddit
              </a>
            </li>
          </ul>
          <button
            type="button"
          >
            Sign Out
          </button>
        </div>
      </section>
    </div>
  </div>
</div>
`;
